[Desmond To Cie Hong] iP

DesmondTo opened the pull request:
### **DukePro**
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/DesmondTo/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the ```main```  method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

DesmondTo committed:
Update .gitignore file

DesmondTo committed:
Add Printer class

DesmondTo committed:
Add greeting message

DesmondTo committed:
Add echo and exit feature

DesmondTo committed:
Implement add-task feature

DesmondTo committed:
Update method name

DesmondTo committed:
Add task-listing method

DesmondTo committed:
Add task-printing feature

DesmondTo committed:
Update task-listing method

DesmondTo committed:
Add Task class

DesmondTo committed:
Update task-listing feature

DesmondTo committed:
Change tasks array's type

DesmondTo committed:
Add status icon in task listing

DesmondTo committed:
Update getStatusIcon method

DesmondTo committed:
Add mark-task-as-done feature

DesmondTo committed:
Add Deadline class

DesmondTo committed:
Add Event class

DesmondTo committed:
Add Todo class

DesmondTo committed:
Simplify Duke.java's code

DesmondTo committed:
Add toString method for Task class

DesmondTo committed:
Update task-listing feature

DesmondTo committed:
Add support for tracking 3 types of tasks

DesmondTo committed:
Debug list command out of bound exception

DesmondTo committed:
Add command inputs

DesmondTo committed:
Add expected output

DesmondTo committed:
Simplify code for Duke.java

DesmondTo committed:
Add DukeException class

DesmondTo committed:
Add EmptyDescriptionException class

DesmondTo committed:
Update delimiter for split's argument

DesmondTo committed:
Handle task with empty description

DesmondTo committed:
Improve empty description error message

DesmondTo committed:
Add UnknownCommandException class

DesmondTo committed:
Handle unknown command exception

DesmondTo committed:
Update storage type of tasks' data>

DesmondTo committed:
Add task-deleting feature

DesmondTo committed:
Update text-ui-testing

DesmondTo committed:
Add enum for user command

DesmondTo committed:
Simplify Duke.java's code

DesmondTo committed:
Add enum for task type

DesmondTo committed:
Add IncompleteDescriptionException class

DesmondTo committed:
Handle task with incomplete description

DesmondTo committed:
Remove redundant code for error handling

DesmondTo committed:
Add javadoc for all the public method

DesmondTo committed:
Update text-ui-test cases

DesmondTo committed:
Update .gitignore

DesmondTo committed:
Update javadoc for public method

DesmondTo committed:
Merge branch 'master' into branch-level-7

DesmondTo committed:
Update 'switch' statement's Java coding standard

DesmondTo committed:
Merge branch 'master' into branch-level-7

DesmondTo committed:
Add method to return data string

DesmondTo committed:
Update toDataString implementation

DesmondTo committed:
Add method to return data string

DesmondTo committed:
Update method's access modifier

DesmondTo committed:
Update 'switch' statement's Java coding standard

DesmondTo committed:
Update task list to global variable

DesmondTo committed:
Merge branch 'master' into branch-level-7

DesmondTo committed:
Remove redundant parameter

DesmondTo committed:
Handle data loading

DesmondTo committed:
Use OS-independent file path

DesmondTo committed:
Handle data file writing

DesmondTo committed:
Resolve file reading bug

DesmondTo committed:
Add InvalidDateFormatException class

DesmondTo committed:
Merge branch 'master' into branch-level-8

DesmondTo committed:
Validate date format in user command

DesmondTo committed:
Remove redundant parameter

DesmondTo committed:
Merge branch 'master' into branch-level-8

DesmondTo committed:
Update .gitignore

DesmondTo committed:
Merge branch 'master' into branch-level-8

DesmondTo committed:
Update date type to LocalDate

DesmondTo committed:
Merge 'branch-level-7' into 'master'

DesmondTo committed:
Merge 'branch-level-8' into 'master'

DesmondTo committed:
Resolve incorrect parameter format

DesmondTo committed:
Add Ui class

DesmondTo committed:
Implement user-interactional methods

DesmondTo committed:
Add Storage class

DesmondTo committed:
Implement methods for Storage class

DesmondTo committed:
Handle directory path more carefully

DesmondTo committed:
Add Parser class

DesmondTo committed:
Add abstract Command class

DesmondTo committed:
Add TaskList class

DesmondTo committed:
Implement method for Command class

DesmondTo committed:
Implement more method for Command class

DesmondTo committed:
Add subclasses of Command class

DesmondTo committed:
Change methods' modifier to public

DesmondTo committed:
Add UnknownCommand class

DesmondTo committed:
Implement method for Parser class

DesmondTo committed:
Refactor the class name

DesmondTo committed:
Update .gitignore

DesmondTo committed:
Update Ui class implementation

DesmondTo committed:
Complete TaskList class implementation

DesmondTo committed:
Complete Storage class implementation

DesmondTo committed:
Update method signature

DesmondTo committed:
Complete UnknownCommand class implementation

DesmondTo committed:
Correct spelling error

DesmondTo committed:
Complete AddCommand class implementation

DesmondTo committed:
Complete ListCommand class implementation

DesmondTo committed:
Complete ExitCommand class implementation

DesmondTo committed:
Complete DoneCommand class implementation

DesmondTo committed:
Complete DeleteCommand class implementation

DesmondTo committed:
Complete Parser class implementation

DesmondTo committed:
Refactor Duke.java

DesmondTo committed:
Update text-ui-testing

DesmondTo committed:
Update catch statement

DesmondTo committed:
Delete redundant class

DesmondTo committed:
Update README

DesmondTo committed:
Update text-ui-testing

DesmondTo committed:
Divide classes into multiple packages

DesmondTo committed:
Add Junit test for AddCommand class

DesmondTo committed:
Remove redundant import

DesmondTo committed:
Handle empty command exception

DesmondTo committed:
Add Junit test for Parser class

DesmondTo committed:
Write Javadoc for AddCommand class

DesmondTo committed:
Write Javadoc for Command class

DesmondTo committed:
Write Javadoc for DeleteCommand class

DesmondTo committed:
Remove redundant imports

DesmondTo committed:
Write Javadoc for DoneCommand class

DesmondTo committed:
Write Javadoc for ExitCommand class

DesmondTo committed:
Write Javadoc for ListCommand class

DesmondTo committed:
Write Javadoc for UnknownCommand class

DesmondTo committed:
Write Javadoc for DukeException class

DesmondTo committed:
Write Javadoc for EmptyDescriptionException class

DesmondTo committed:
Add fullstop to Javadoc

DesmondTo committed:
Write Javadoc for IncompleteDescriptionException class

DesmondTo committed:
Write Javadoc for InvalidDateFormatException class

DesmondTo committed:
Write Javadoc for UnknownCommandException class

DesmondTo committed:
Write Javadoc for Deadline class

DesmondTo committed:
Write Javadoc for Event class

DesmondTo committed:
Write Javadoc for Task class

DesmondTo committed:
Write Javadoc for Todo class

DesmondTo committed:
Correct Javadoc's error

DesmondTo committed:
Write Javadoc for Parser class

DesmondTo committed:
Write Javadoc for Storage class

DesmondTo committed:
Write Javadoc for TaskList class

DesmondTo committed:
Write Javadoc for Ui class

DesmondTo committed:
Write Javadoc for Duke class

DesmondTo committed:
Add white space at left side of curly braces

DesmondTo committed:
Shorten code to less then 120 char per line

DesmondTo committed:
Add FindCommand class

DesmondTo committed:
Merge branch 'branch-A-JavaDoc' into 'master'

DesmondTo committed:
Resolve merge conflict

DesmondTo committed:
Add curly braces for single 'if' statement

DesmondTo committed:
Add task-finding feature

DesmondTo committed:
Update print format for search result

DesmondTo committed:
Merge branch 'branch-level-9' into 'master'

DesmondTo committed:
Complete task-finding feature

DesmondTo committed:
Merge pull request #1 from DesmondTo/branch-Level-9

Branch level 9

willyamped commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /**
+     * Execute the chatbot functions based on user's command.
```
I believe Javadoc headers should start with words in the form `Returns ...`, `Sends ...`, `Adds ...` (not `Return` or `Returning` etc.)
This applies to your other javadocs headers  too :)

willyamped commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,117 @@
+
+    private Task createTask(String description, LocalDate date) {
+
+        switch (TaskType.valueOf(this.taskType)) {
```
I believe you should avoid unnecessary use of `this`. Instead, you could just write `TaskType.valueOf(taskType)`. This applies to your other codes as well :)

willyamped commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+public class TaskList {
+
+    private enum TaskType {
+        TODO, EVENT, DEADLINE
```
I like how you make use of `enum` for this 👍

willyamped commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,105 @@
+     */
+    public void add(Task task) {
+        this.tasks.add(task);
+        this.taskNum++;
```
Similar to my previous comment. I believe you should avoid unnecessary use of `this` :)

willyamped reviewed (commented):
LGTM. The code looks good. The only problem is with the javadocs headers and the use of unnecessary `this.`

DesmondTo committed:
Merge branch 'add-gradle-support' of iP

DesmondTo committed:
Add gradle checkstyles

DesmondTo committed:
Merge pull request #2 from DesmondTo/branch-A-Gradle

Add gradle checkstyles

DesmondTo committed:
Revert "Merge pull request #2 from DesmondTo/branch-A-Gradle"

This reverts commit 5c1f31706d232bf2233001521553dcd84822db5f, reversing
changes made to bfc77c159f5a1e621dfd0c1ecc0558f7fa8e3462.

DesmondTo committed:
Revert "Add gradle checkstyles"

This reverts commit acec83662c9b374ffb1f65b4c00237f82656a5d0.

DesmondTo committed:
Modify application mainClassName

DesmondTo committed:
Merge pull request #3 from DesmondTo/branch-A-Gradle

Branch a gradle

DesmondTo committed:
Add checkstyle.xml

DesmondTo committed:
Add suppressions.xml

DesmondTo committed:
Merge pull request #4 from DesmondTo/branch-A-CheckStyle

Branch a check style

DesmondTo committed:
Update dependencies

DesmondTo committed:
Add Launcher class

DesmondTo committed:
Add empty constructor

DesmondTo committed:
Update font of label

DesmondTo committed:
Update layout

DesmondTo committed:
Add handler methods

DesmondTo committed:
Remedy the confines of the ScrollPane

DesmondTo committed:
Add custom control DialogBox

DesmondTo committed:
Remove redundant imports

DesmondTo committed:
Handle user input

DesmondTo committed:
Add custom behavior to DialogBox

DesmondTo committed:
Use FXML for Duke

DesmondTo committed:
Refactor code for command and response

DesmondTo committed:
Merge pull request #5 from DesmondTo/branch-Level-10

Branch level 10

DesmondTo committed:
Resolve coding standard violations

DesmondTo committed:
Resolve coding standard violations

DesmondTo committed:
Remove redundant line

DesmondTo committed:
Update response format

luffingluffy commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,55 @@
+    private final int taskNum;
+
+    /**
+     * Construcyts a DeleteCommand object that handles task-deletion command.
```
I think there's a typo here.

luffingluffy reviewed (commented):
Well-written and well-documented code with no formatting issues. LGTM!

DesmondTo committed:
Improve code quality

DesmondTo committed:
Restore Duke main method

DesmondTo committed:
Revert "Restore Duke main method"

This reverts commit 59fbe06ccb5992020ef8f46d2a7d4f484bcd7325.

DesmondTo committed:
Improve GUI

DesmondTo committed:
Update archiveClassifier

DesmondTo committed:
Add assertions

Some assumptions has been made but missing assertions in the functional
code (eg. taskNum >= 1).

Some of these assumptions cause program malfunction.

Assertions should be used to verify assumptions about the program to
detect potential bugs and terminate the program as soon as possible
upon assertion failures.

Let's add assertions in the functional code.

Using assertions worth the additional safety they provide.

Refers to this notes on using assertions
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.htm
l#w5-6-assertions

DesmondTo committed:
Improve code quality

Many methods can be written using KISS principle and SLAP principle.

Long methods can be difficult to debug and reduce readability.

Extracting the part of the code into a new method
with shorter implementation. Also, keep a consistent level of
abstraction within a code fragment

Let's apply KISS principle and SLAP principle.

Using KISS principle and SLAP principle are preferred to
improve the readability of the code.

Refer to this notes on code quality.
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.htm
l#w5-4-code-quality

DesmondTo committed:
Merge pull request #6 from DesmondTo/branch-A-Assertions

Add assertions

DesmondTo committed:
Merge branch 'master' into branch-A-CodeQuality

DesmondTo committed:
Merge pull request #7 from DesmondTo/branch-A-CodeQuality

Improve code quality

DesmondTo committed:
Set up Continuous Integration

DesmondTo committed:
Add window I/O redirection tests

DesmondTo committed:
Update gradle.yml

DesmondTo committed:
Improve code style

DesmondTo committed:
Update JUnit test

DesmondTo committed:
Remove Window I/O redirection tests

DesmondTo committed:
Add task-tagging feature

DesmondTo committed:
Merge pull request #8 from DesmondTo/branch-C-Tagging

Add task-tagging feature

DesmondTo committed:
Improve code quality

DesmondTo committed:
Rearrange methods' order

DesmondTo committed:
Update icon and product name

DesmondTo committed:
Disable resizability

DesmondTo committed:
Update basic UI

DesmondTo committed:
Update app basic UI

DesmondTo committed:
Update dialog box UI

DesmondTo committed:
Update chatbot personality

DesmondTo committed:
Set theme jekyll-theme-midnight

DesmondTo committed:
Update README

DesmondTo committed:
Merge branch 'master' of https://github.com/DesmondTo/ip

DesmondTo committed:
Update README

DesmondTo committed:
Update README

DesmondTo committed:
Update README section link

DesmondTo committed:
Update README section link

DesmondTo committed:
Update README

DesmondTo committed:
Remove redundant whitespace

DesmondTo committed:
Handle more exceptions

DesmondTo committed:
Split AddCommand into three classes

DesmondTo committed:
Add utility classes

DesmondTo committed:
Update Task Class and its subclasses

DesmondTo committed:
Update utility class

DesmondTo committed:
Update error message pattern

DesmondTo committed:
Update command classes

DesmondTo committed:
Update error message style

DesmondTo committed:
Update exception testings

DesmondTo committed:
Add next line

DesmondTo committed:
Update Javadoc parameter

DesmondTo committed:
Resolve minor searching bug
